### PR TITLE
async user token request

### PR DESCRIPTION
**Pull request by pletnes**

### CODE REVIEW

**Comment by pletnes** [comment]
@robertofalk this implements an async variant of the `request_token_for_client` method. It is implemented alongside the sync method, so it should not break any previous behavior of the code, as far as I can tell.

**Comment by robertofalk** [comment]
it look ok, to me, only the version file is missing. I think you can increase the minor since you are adding in a compatible way.

**Comment by pletnes** [comment]
@robertofalk please review again - I've adapted this PR to changes in #54,  hope I get everything right!

I kept the changelog short and sweet. If you want more details there, just let me know, and I can elaborate more about the rationale.

**Comment by pletnes** [comment]
@robertofalk I notice that Travis CI does not give a status - the build seems to be created, but does not run. Is there anything I should or could do about this?

**Comment by robertofalk** [comment]
> @robertofalk I notice that Travis CI does not give a status - the build seems to be created, but does not run. Is there anything I should or could do about this?

I think PRs created in the old travis were not migrated. I'm running the tests locally, then I merge if it's all green.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,11 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+## 3.1.0
```
Comment (robertofalk) [inline CHANGELOG.md:8]: This has to be 3.2.0, since 3.1.0 was already released

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,11 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+## 3.2.0
```
Comment (pletnes) [inline CHANGELOG.md:8]: Better now @robertofalk ?

File: CHANGELOG.md (hunk 1 above)
Comment (pletnes) [reply CHANGELOG.md]: Fixed, see new commit.

**Review by robertofalk — APPROVED** [review]

**Pull request merged**